### ISSUE TITLE
[Halaa Battle tokens] Still not correctly given - Nagrand

### ISSUE THREAD

**Issue by Echinaceax** [WoWManiaUK/Redemption#2989]
**_Halaa Battle tokens_**

**_Reported by Snootsay on redemption
tested by me and them_**  

> 1:16 AM] Scootsay: You acquire a Halaa Battle Token by being the member of a group/raid that gets a killing blow on an enemy player in the vicinity of Halaa in Nagrand. This includes the main city itself and all the way out to the wyvern roosts surrounding the city. When the enemy player is killed (and your group gets the killing blow), one of these tokens will appear in your inventory if you are in range.
> [1:16 AM] Scootsay: I did, but its suppose to work off every kill, even before fighting the guards it should of worked

- Tested by me (horde) Scootsay (ally) 
- Didnt yeald token while Horde guards were alive. 
- Only worked when Flag was "contested"

- tested token drops while stood at the contested flag, both was pvp enabled
- then tokens where "made by ourselves"

- Scootsay killed me (he/she) made there own token 
- I killed Scootsay and I made own token 
- **player makes own token by killing the opposite faction**
- they drop in our bag after kills 

![we make them](https://user-images.githubusercontent.com/39439201/51288088-d5bd9280-19fa-11e9-957e-ce1d6823f8e7.jpg)

- Now the problem is - is that tokens are meant to work in and around Halaa, even if flag isnt contested for horde or for Alliance.
- How it works atm is that players have to have the flag "contested" first.
- It shouldnt be that way

**Comment by wow-mania** [comment]
The tokens are given at any time as long as it is an honorable kill.

**Comment by NightSeekerr** [comment]
Actually this is still broken: 
![WoWScrnShot_033019_012819](https://user-images.githubusercontent.com/30464528/55268718-18220c00-5295-11e9-910f-3f4fb94d9922.jpg)

I kill an enemy player that wields honor in Halaa and I only recieve Honor Points, no Halaa Battle Tokens.   I've tested this more than once btw. I use the latest client version.

**Comment by Scootsay** [comment]
I finally got mine farmed out, you have to kill the guards and have enemy faction there and flagged for pvp. When someone acquires halaa and spawn guards. Then you can farm them as much as they will allow you to. Only way I could get it to work

**Comment by NightSeekerr** [comment]
Will this ever get fixed or what's going on?

**Comment by Echinaceax** [comment]
@NightSeekerr 
Iam wondering the same thing.

wow-mania closed this because there wasnt enough info given, i tried to explain but he didnt understand to "what was happening" and what should happen"

> The tokens are given at any time as long as it is an honorable kill. Kill an honorable player at any time and you will get the token.

if you can reword it to show "what is happening" and what should happen" it would help wow-mania to reproduce

**Comment by Dranah** [comment]
From PTR: Killed ally with horde, then horde with ally. Both got tokens, flag is owned by Horde.
![WoWScrnShot_040719_214504](https://user-images.githubusercontent.com/31132743/55688907-8db96680-597e-11e9-8feb-81745ff64993.jpg)
![WoWScrnShot_040719_214429](https://user-images.githubusercontent.com/31132743/55688908-914ced80-597e-11e9-84f2-67760ad37cf5.jpg)
What did you do different @NightSeekerr ?

**Comment by wow-mania** [comment]
I will fix the health

**Comment by Dranah** [comment]
Ok, I found the issue.
Sometimes the status bar on the right side of screen disappear. Usually when you relog or arrive late to the party (halaa is already captured and you didnt take part in it - entering the area and then enabling pvp won't automatically pop the bar)
Now, kills made in this state (not seeing the status bar) do not award a Halaa Battle tokens somehow.
![WoWScrnShot_040719_215946](https://user-images.githubusercontent.com/31132743/55689228-4fbe4180-5982-11e9-8bdb-cc3f7d9afd71.jpg)

The way it interacts with what Echi described (needing the base to be contested) is that you are only able to re-acquire your status bar when it is able to move. When the base is contested, turning pvp on and off or attacking an enemy will pop up your status bar. If one faction controls firmly, however, only by killing guards I managed to re-acquire it.

**Comment by wow-mania** [comment]
I fixed the NPC and the notification of state as much as possible. Nothing much to be done without risking messing it up.

**Comment by Echinaceax** [comment]
> I fixed the NPC and the notification of state as much as possible. Nothing much to be done without risking messing it up.

Thanks Jani :)

**Comment by Dranah** [comment]
Health of alliance guards fixed (same 590k as horde now)
Still got issue with sometimes not being able to trigger the status bar. Kills in this state are not giving tokens.
The only way to fix it is to overtake the base with opposite faction

**Comment by NightSeekerr** [comment]
So, the main issue still exists. I cannot get a Halaa Battle Token when I kill an enemy player that gives Honor  in Halaa when Halaa is not contested (which is 99% of the time) and when it's contested it stays contested for a minute or two until it's taken by one of the factions again. Can this be fixed?

**Comment by Dranah** [comment]
You get the token as intended when the status bar is visible in whatever state.
